assets/app/scripts/directives/quotaUsageChart.js

```diff
@@ -0,0 +1,122 @@
+"use strict";
+
+angular.module('openshiftConsole')
+  .directive('quotaUsageChart', function($filter, Logger) {
+    return {
+      restrict: 'E',
+      scope: {
+        used: '=',
+        total: '=',
+        // 'cpu' or 'memory'
+        type: '@',
+        // Defaults to 'bottom'.
+        // http://c3js.org/reference.html#legend-position
+        legendPosition: '@?'
+      },
+      // Replace the element so it can be centered using class="center-block".
+      replace: true,
+      templateUrl: 'views/_quota-usage-chart.html',
+      link: function($scope, element) {
+        var usageValue = $filter('usageValue');
+        var usageWithUnits = $filter('usageWithUnits');
+        function updateCenterText() {
+          var donutChartTitle = d3.select(element[0]).select('text.c3-chart-arcs-title');
+          if (!donutChartTitle) {
+            Logger.warn("Can't select donut title element");
+            return;
+          }
+
+          var valueAndType = usageWithUnits($scope.total, $scope.type).split(' ');
+          var largeText = valueAndType[0];
+          var smallText = valueAndType[1];
+
+          // Replace donut title content.
+          donutChartTitle.selectAll('*').remove();
+          donutChartTitle.insert('tspan').text(largeText).classed('pod-count donut-title-big-pf', true).attr('dy', 0).attr('x', 0);
+          donutChartTitle.insert('tspan').text(smallText).classed('donut-title-small-pf', true).attr('dy', 20).attr('x', 0);
+        }
+
+        // Adjust size based on legend position.
+        if ($scope.legendPosition === 'right') {
+          $scope.height = 175;
+          $scope.width = 250;
+        } else {
+          $scope.height = 200;
+          $scope.width = 175;
+        }
+
+        // https://github.com/mbostock/d3/wiki/Formatting
+        var percentage = d3.format(".2p");
+
+        // Chart configuration, see http://c3js.org/reference.html
+        $scope.chartID = _.uniqueId('quote-usage-chart-');
+        var config = {
+          type: "donut",
+          bindto: '#' + $scope.chartID,
+          donut: {
+            label: {
+              show: false
+            },
+            width: 10
+          },
+          size: {
+            height: $scope.height,
+            width: $scope.width
+          },
+          legend: {
+            show: true,
+            position: $scope.legendPosition || 'bottom',
+            item: {
+              // Don't hide arcs when clicking the legend.
+              onclick: _.noop
+            }
+          },
+          onrendered: updateCenterText,
+          tooltip: {
+            position: function() {
+              // Position in the top-left to avoid problems with tooltip text wrapping.
+              return { top: 0, left: 0 };
+            },
+            format: {
+              value: function(value) {
+                return percentage(value / usageValue($scope.total)) +
```

Comment:
if quota is set to 0 for something (unrealistic for these particular resource types, but i think possible), this would NaN not sure what percentage does on a NaN